### PR TITLE
Create gossip-findings-postgres crate with error types and conversion

### PR DESCRIPTION
**Pull request by ahrav**
helpers

Scaffold the findings PostgreSQL persistence crate with three core modules:

- `error.rs`: FindingsPgSchemaError (contracts + conversion failures) and FindingsPgMigrationError (driver/SQL + checksum integrity)
- `types.rs`: u64 ↔ BIGINT conversion helpers supporting two storage modes: bit-pattern (full u64 range, equality/grouping only) and ordered non-negative (0..=i64::MAX, preserves SQL ordering)
- `schema.rs` and `migrations.rs`: Public namespaces for schema constants and migration runners (definitions follow in D1.2 and D1.3)

All error types implement Display, Error, and Debug. Conversion helpers are
inline with comprehensive boundary-value and property-based tests.

## Summary by CodeRabbit

* **New Features**
  * Added PostgreSQL findings backend scaffold with schema and migration support
  * Added safe bidirectional u64↔BIGINT conversion utilities
  * Added richer, descriptive error types for database interactions

* **Tests**
  * Added unit, property-based and fuzzing tests to validate conversions, migrations, and error behavior

* **Chores**
  * Updated architecture and scope documentation to include the new PostgreSQL primitives and findings scaffold

### CODE REVIEW

File: crates/gossip-findings-postgres/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        Self::PgU64Conversion(value)
+    }
+}
+
+/// Labels the migration step that produced a PostgreSQL driver error.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum MigrationOperation {
+    /// Initial TCP/TLS connection to the database.
+    Connect,
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-findings-postgres/src/error.rs]: Agreed — `MigrationOperation` is now defined once in `gossip-pg-common::migration` and re-exported by both crates via `pub use`. The duplicate enum + Display impl have been removed from both `error.rs` files.

File: crates/gossip-findings-postgres/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        let err = FindingsPgSchemaError::from(PgU64ConversionError::NegativeStoredValue {
+            field: "byte_offset",
+            value: -1,
+        });
+
+        let source = err.source().expect("schema error should expose source");
+        assert_eq!(
+            source.to_string(),
+            "stored value -1 for field byte_offset is negative but the schema requires non-negative BIGINT"
+        );
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-findings-postgres/src/error.rs]: Good point — these tests verify delegation, not upstream wording. Refactored all four tests to compare against `inner.to_string()` instead of hardcoded string literals. Now an upstream Display change won't break these tests as long as the delegation still works.
